tests/integration/test_authd/test_authd_agents_ctx.py

```diff
@@ -0,0 +1,425 @@
+    - CentOS 7
+    - Debian Buster
+    - Red Hat 8
+    - Ubuntu Focal
+    - Ubuntu Bionic
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/reference/daemons/wazuh-authd.html
+    - https://documentation.wazuh.com/current/user-manual/reference/tools/agent_groups.html
+
+tags:
+    - enrollment
+'''
+import os
+import subprocess
+import time
+from pathlib import Path
+
+import pytest
+from wazuh_testing.constants.paths.logs import WAZUH_PATH, WAZUH_LOG_PATH
+from wazuh_testing.constants.paths.sockets import AUTHD_SOCKET_PATH
+from wazuh_testing.constants.paths.configurations import WAZUH_CLIENT_KEYS_PATH
+from wazuh_testing.utils.file import truncate_file, remove_file, recursive_directory_creation
+from wazuh_testing.tools.monitors import file_monitor
+from wazuh_testing.utils.services import control_service, check_daemon_status
+from wazuh_testing.modules.api.utils import remove_groups, set_up_groups
+from wazuh_testing.tools.wazuh_manager import remove_all_agents
+
+from . import CONFIGURATIONS_FOLDER_PATH, TEST_CASES_FOLDER_PATH
+from wazuh_testing.utils.configuration import get_test_cases_data, load_configuration_template
+
+# Marks
+
+pytestmark = [pytest.mark.linux, pytest.mark.tier(level=0), pytest.mark.server]
+
+# Paths
+test_configuration_path = Path(CONFIGURATIONS_FOLDER_PATH, 'config_authd_common.yaml')
+test_cases_path = Path(TEST_CASES_FOLDER_PATH, 'cases_authd_agents_ctx.yaml')
+
+# Configurations
+test_configuration, test_metadata, test_cases_ids = get_test_cases_data(test_cases_path)
+test_configuration = load_configuration_template(test_configuration_path, test_configuration, test_metadata)
+
+# Variables
+daemons_handler_configuration = {'all_daemons': True}
+
+receiver_sockets_params = [(("localhost", 1515), 'AF_INET', 'SSL_TLSv1_2'), (AUTHD_SOCKET_PATH, 'AF_UNIX', 'TCP')]
+
+receiver_sockets = None  # Set in the fixtures
+groups_infra = ['001','002', '003', '004']
+test_group = "TestGroup"
+timeout = 10
+login_attempts = 3
+sleep = 1
+
+def clean_agents_ctx():
+    clean_keys()
+    clean_rids()
+    clean_agents_timestamp()
+    clean_diff()
+
+
+def wait_server_connection():
+    """Wait until agentd has begun"""
+
+    def callback_agentd_startup(line):
+        if 'Accepting connections on port 1515' in line:
+            return line
+        return None
+
+    log_monitor = file_monitor.FileMonitor(WAZUH_LOG_PATH)
+    log_monitor.start(timeout=30, callback=callback_agentd_startup)
+
+
+def clean_logs():
+    truncate_file(WAZUH_LOG_PATH)
+
+
+def clean_keys():
+    truncate_file(WAZUH_CLIENT_KEYS_PATH)
+
+
+def clean_diff():
+    diff_folder = os.path.join(WAZUH_PATH, 'queue', 'diff')
+    try:
+        remove_file(diff_folder)
+        recursive_directory_creation(diff_folder)
+        os.chmod(diff_folder, 0o777)
+    except Exception as e:
+        print('Failed to delete %s. Reason: %s' % (diff_folder, e))
+
+
+def clean_rids():
+    rids_folder = os.path.join(WAZUH_PATH, 'queue', 'rids')
```

Comment:
Add this queue to the framework in `constants/paths/sockets.py` and use it. Modify this where needed.